.github/workflows/build.yml

```diff
@@ -25,13 +26,11 @@ jobs:
     - run: sudo apt-get install ccache
     - run: make setup
     - run: make -j $(nproc) simulator-tests
-  check-astyle:
+  check-formatting:
     runs-on: ubuntu-latest
     steps:
-    - run: sudo apt-get install astyle
     - uses: actions/checkout@v2
-    - run: make setup
-    - run: make check-astyle
+    - run: bin/format-code.sh --check
```

Comment:
You don't need to run 'make setup', but I'd rather we stick to the makefile `check-formatting` target in the github action. 

The perf issue you flagged is real, but somewhat immaterial for running in CI

Reply:
I do have an aversion to using makefiles for anything other than compiling, but that's mostly because the syntax is relatively unfamiliar (and therefore bizarre and mysterious).  If you can make it so `make check-formatting` doesn't fail without first running `make setup`, that's fine with me.  The reason I did it this was was that I _did_ need the `make setup`, or the other target wouldn't run properly.

Reply:
yup https://github.com/keyboardio/Kaleidoscope/tree/f/no-make-implicit-rules is poorly named but has that fix